src/core/reaction_ensemble.cpp

```diff
@@ -1,1724 +1,1970 @@
 }
 
 /**
-* Replaces a particle with the given particle id to be of a certain type. This especially means that the particle type and the particle charge are changed.
+* Hides a particle from short ranged interactions and from the electrostatic
+* interaction. Additional hiding from interactions would need to be implemented
+* here.
 */
-int ReactionEnsemble::replace(int p_id, int desired_type){
-	int err_code_type=set_particle_type(p_id, desired_type);
-	int err_code_q = 0.0;
-	#ifdef ELECTROSTATICS
-	err_code_q=set_particle_q(p_id, (double) m_current_reaction_system.charges_of_types[find_index_of_type(desired_type)]);
-	#endif
-	return (err_code_q bitor err_code_type);
-}
-
+int ReactionAlgorithm::hide_particle(int p_id, int previous_type) {
 /**
-* Hides a particle from short ranged interactions and from the electrostatic interaction. Additional hiding from interactions would need to be implemented here.
+*remove_charge and put type to a non existing one --> no interactions anymore it
+*is as if the particle was non existing (currently only type-based interactions
+*are swithced off, as well as the electrostatic interaction)
+*hide_particle() does not break bonds for simple reactions. as long as there are
+*no reactions like 2A -->B where one of the reacting A particles occurs in the
+*polymer (think of bond breakages if the monomer in the polymer gets deleted in
+*the reaction). This constraint is not of fundamental reason, but there would be
+*a need for a rule for such "collision" reactions (a reaction like the one
+*above).
 */
-int ReactionEnsemble::hide_particle(int p_id, int previous_type){
-	/**
-    *remove_charge and put type to a non existing one --> no interactions anymore it is as if the particle was non existing (currently only type-based interactions are swithced off, as well as the electrostatic interaction)  
-    *hide_particle() does not break bonds for simple reactions. as long as there are no reactions like 2A -->B where one of the reacting A particles occurs in the polymer (think of bond breakages if the monomer in the polymer gets deleted in the reaction). This constraint is not of fundamental reason, but there would be a need for a rule for such "collision" reactions (a reaction like the one above).
-    */
-	#ifdef ELECTROSTATICS
-	//set charge
-	set_particle_q(p_id, 0.0);
-	#endif
-	//set type
-	int err_code_type=set_particle_type(p_id, m_current_reaction_system.non_interacting_type);
-	return err_code_type;
+#ifdef ELECTROSTATICS
+  // set charge
+  set_particle_q(p_id, 0.0);
+#endif
+  // set type
+  int err_code_type =
+      set_particle_type(p_id, non_interacting_type);
+  return err_code_type;
 }
 
 /**
-* Deletes the particle with the given p_id. This method is intended to only delete unbonded particles since bonds are coupled to ids. The methods is aware of holes in the particle id range and fills them if a particle gets created.
+* Deletes the particle with the given p_id. This method is intended to only
```

Comment:
If you write function documentation, please use the doxygen syntax in all places.